src/common/type/stringId.c

```diff
@@ -0,0 +1,375 @@
+                    if (size >= STRID6_MAX)
+                        result |= STRING_ID_PREFIX;
+
+                    result |= (uint64_t)map[(uint8_t)buffer[9]] << 58;
+                }
+
+                case 9:
+                    result |= (uint64_t)map[(uint8_t)buffer[8]] << 52;
+
+                case 8:
+                    result |= (uint64_t)map[(uint8_t)buffer[7]] << 46;
+
+                case 7:
+                    result |= (uint64_t)map[(uint8_t)buffer[6]] << 40;
+
+                case 6:
+                    result |= (uint64_t)map[(uint8_t)buffer[5]] << 34;
+
+                case 5:
+                    result |= (uint64_t)map[(uint8_t)buffer[4]] << 28;
+
+                case 4:
+                    result |= (uint64_t)map[(uint8_t)buffer[3]] << 22;
+
+                case 3:
+                    result |= (uint64_t)map[(uint8_t)buffer[2]] << 16;
+
+                case 2:
+                    result |= (uint64_t)map[(uint8_t)buffer[1]] << 10;
+
+                case 1:
+                    result |= (uint64_t)map[(uint8_t)buffer[0]] << 4;
+            }
+
+            FUNCTION_TEST_RETURN(result);
+        }
+    }
+}
+
+/**********************************************************************************************************************************/
+size_t
+strIdToZN(StringId strId, char *const buffer)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(STRING_ID, strId);
+        FUNCTION_TEST_PARAM(VOID, buffer);
+    FUNCTION_TEST_END();
+
+    ASSERT(strId != 0);
+    ASSERT(buffer != NULL);
+
+    // Is the StringId a prefix of a longer string?
+    bool prefix = strId & STRING_ID_PREFIX;
+
+    // Extract bits used to encode the characters
+    StringIdBit bit = (StringIdBit)(strId & STRING_ID_BIT_MASK);
+
+    // Remove header to get the encoded characters
+    strId >>= STRING_ID_HEADER_SIZE;
+
+    // Decoding type
+    switch (bit)
+    {
+        // 5-bit decoding
+        case stringIdBit5:
+        {
+            // Map to convert encoding to characters
+            const char map[32] = "!abcdefghijklmnopqrstuvwxyz-1234";
+
+            // Macro to decode all but the last character
+            #define STR5ID_TO_ZN_IDX(idx)                                                                                          \
+                buffer[idx] = map[strId & 0x1F];                                                                                   \
+                strId >>= 5;                                                                                                       \
+                                                                                                                                   \
+                if (strId == 0)                                                                                                    \
+                    FUNCTION_TEST_RETURN(idx + 1)
+
+            // Char 1-11
+            STR5ID_TO_ZN_IDX(0);
```

Comment:
Can't we do this in a loop and use the STRID5_MAX like 
```
for (unsigned int idx = 0, idx < STRID5_MAX - 3; idx++)
    STR5ID_TO_ZN_IDX(idx);
```

Reply:
Yes, but this is much faster.